File: Sources/Scan.swift
import Foundation

func runScan(args: [String]) -> Int32 {
    var seconds: TimeInterval = 5.0
    var i = 0
    while i < args.count {
        switch args[i] {
        case "--seconds":
            guard i + 1 < args.count, let v = TimeInterval(args[i + 1]) else {
                FileHandle.standardError.write(Data("usage: eegcli scan [--seconds N]\n".utf8))
                return 2
            }
            seconds = v; i += 2
        case "-h", "--help":
            print("usage: eegcli scan [--seconds N]")
            return 0
        default:
            FileHandle.standardError.write(Data("unknown arg: \(args[i])\n".utf8))
            return 2
        }
    }

    print("Scanning for \(Int(seconds))s...")
    let (_, sensors) = scanForBrainBits(seconds: seconds)
    if sensors.isEmpty {
        print("No headbands found.")
        print("- Make sure the device is powered on and in range.")
        print("- Approve the macOS Bluetooth permission prompt for your terminal.")
        return 1
    }
    print("Found \(sensors.count) device(s):")
    for s in sensors {
        print("  - \(s.name)  serial=\(s.serialNumber)  addr=\(s.address)  rssi=\(s.rssi) dBm")
    }
    return 0
}

File: src/scan.py
"""Discover BrainBit headbands within Bluetooth range."""
from __future__ import annotations

import argparse

from src._device import scan


def main() -> int:
    ap = argparse.ArgumentParser(description="Scan for BrainBit headbands.")
    ap.add_argument("--seconds", type=float, default=5.0, help="scan duration (default 5)")
    args = ap.parse_args()

    print(f"Scanning for {args.seconds:.0f}s...")
    scanner, sensors = scan(args.seconds)

    if not sensors:
        print("No headbands found. Make sure the device is powered on and in range,")
        print("and that Terminal has Bluetooth permission in System Settings.")
        return 1

    print(f"Found {len(sensors)} device(s):")
    for s in sensors:
        print(f"  - {s.Name}  serial={s.SerialNumber}  addr={s.Address}  rssi={s.RSSI} dBm")

    del scanner
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
